validity_rollups_on_bitcoin.md

```diff
@@ -280,7 +325,7 @@ The onchain transactions needed to open and settle (and occasionally rebalance)
 
 The ability to layer validity rollups on top of one another also means that capacity for Lightning can be added as-needed. Each time a rollup reaches its maximum capacity, another rollup could be deployed as an additional layer, then Lightning onboarding can continue until that layer fills up, and so on.
 
-Although Lightning is most famous for settling on the bitcoin mainchain, Lightning channels can be opened on any Lightning-compatible blockchain. Additionally, if there is a Lightning node with channels open on two or more chains, it can route payments between users of these different chains, as if those users had channels open on the same chain. In cases where there are different assets being sent and received through Lightning channel routes (e.g. BTC and LTC) there is an "inadvertent call option problem".[88] However, a Lightning channel route that uses the same asset throughout the entire route, even if the route crosses multiple chains, does not suffer from this problem, or at least the problem is negligible when accounting for any small price differences between the different versions of the same asset on different chains. So users who have Lightning BTC channels open on bitcoin Layer 1 could seamlessly transact with users who have Lightning BTC channels open on Layer 2+ validity rollups provided there is a route with sufficient liquidity between the sender and recipient.
+Although Lightning is most famous for settling on the bitcoin mainchain, Lightning channels can be opened on any Lightning-compatible blockchain. Additionally, if there is a Lightning node with channels open on two or more chains, it can route payments between users of these different chains, as if those users had channels open on the same chain. In cases where there are different assets being sent and received through Lightning channel routes (e.g. BTC and LTC) there is an "inadvertent call option problem".[97] However, a Lightning channel route that uses the same asset throughout the entire route, even if the route crosses multiple chains, does not suffer from this problem, or at least the problem is negligible when accounting for any small price differences between the different versions of the same asset on different chains. So users who have Lightning BTC channels open on bitcoin L1 could seamlessly transact with users who have Lightning BTC channels open on L2+ validity rollups provided there is a route with sufficient liquidity between the sender and recipient.
```

Comment:
Not sure if you want to go into it much but it could be interesting to mention that in a lightning-oriented rollup the open/close L1 data costs could be much smaller, beyond just the lack of signatures.

Reply:
> it could be interesting to mention that in a lightning-oriented rollup the open/close L1 data costs could be much smaller, beyond just the lack of signatures.

how so?

Reply:
If we assume that "funds at rest" are stored in an account-like structure then a channel open would just involve 8 bytes for the indexes of two parties, plus you could do 2 bytes for the channel size (we can get by with a fixed list of sizes), and maybe plus ~2 more bytes for any metadata and fee information.  We can assume either even dualfunding or the first of the two parties fronts the full balance.

Storing channels could be done either with sequential indexes or some commitment to each party.  In the worst case, close txs would involve 8 bytes for the two indexes of each party again, the ~2 bytes for the meta+fee, probably 4 bytes for the end balance, and then for each HTLC 20 bytes for the hash, 4 bytes for the balance and direction, and 4 bytes for the timeout.  HTLCs can get stored in their own indexes table and remember who is allowed to claim them so that information doesn't need to be included in the claim tx.

The HTLC claims are just the 4 byte index of the HTLC, a 20 byte preimage, and probably another ~2 bytes for meta+fee.  It's possible you could even make timed-out HTLCs reclaim themselves automatically.

Reply:
Ah yes, I had considered including an account-based version of the Lightning rollup for completeness, but wasn't sure about the transaction size details. So Table 3 and 4 would be modified to look like this:

#### Table 3. Comparing the size of different types of UTXO model dual-funded Lightning channel open transactions

| Transaction type                       | Non-discounted data weight (4x) | Discounted data weight (1x) | Total (WU) |
| -------------------------------------- |:-------------------------------:|:---------------------------:|:----------:|
| Rollup LN channel open (account model) | 0                               | 12                          | 12         |
| Rollup LN channel open (UTXO model)    | 0                               | 197.5                       | 197.5      |
| Mainchain LN channel open              | 795                             | 211                         | 1006       |

> Table 3 description: The first rollup transaction is a dual-funded Lightning channel open transaction using an account model rollup. The second rollup transaction is a 2-P2WPKH-input-1-P2WSH-output-2-P2WPKH-output transaction with the 27 byte witnesses stripped from each input. The mainchain transaction is a 2-P2WPKH-input-1-P2WSH-output-2-P2WPKH-output transaction.[^96]

#### Table 4. Comparing the per-block capacity of different types of UTXO model dual-funded Lightning channel open transactions

| Transaction type                       | 1,000,000 WU                          | 3,000,0000 WU            | Total (tx/block) |
| -------------------------------------- |:-------------------------------------:|:------------------------:|:----------------:|
| Rollup LN channel open (account model) | Validity proof, script, other tx data | 250,000 txs (12 WU/tx)   | 250,000          |
| Rollup LN channel open (UTXO model)    | Validity proof, script, other tx data | 15,190 txs (197.5 WU/tx) | 15,190           |
| Mainchain LN channel open              | 994 txs (1006 WU/tx)                  | 2,982 txs (1006 WU/tx)   | 3,976            |

> Table 4 description: The block is broken up into 1,000,000 WU and 3,000,000 WU portions to illustrate how in the case of rollup transactions, some of the weight limit is reserved for the validity proof and script data while the rest can be used for the rollup transaction data.

Is that right @delbonis?